System/OsPath/Data/ByteString/Short/Word16.hs

```diff
@@ -647,6 +650,23 @@ splitWith p = \(assertEven -> sbs) -> if
               | otherwise -> a : go (tail b)
 
 
+-- | Check whether one string is a substring of another.
+isInfixOf :: ShortByteString -> ShortByteString -> Bool
+isInfixOf sbs = \s -> null sbs || not (null $ snd $ GHC.Exts.inline breakSubstring sbs s)
+
+
+breakSubstring :: ShortByteString -- ^ String to search for
+               -> ShortByteString -- ^ String to search in
+               -> (ShortByteString, ShortByteString) -- ^ Head and tail of string broken at substring
+breakSubstring pat inp = go 0 inp
+ where
+    go prefix s = let (h, t)  = BS.breakSubstring pat s
+                      hl      = BS.length h
+                  in if | even hl   -> (BS.take (hl + prefix) inp, t)
```

Comment:
Another thing that you could do  is to optimize the case where we are done while `prefix` is still `0`. Here you could just return `(h, t)` without any additional `take`. Not sure if it's worth the extra complexity.